blivet/tasks/availability.py

```diff
@@ -117,71 +118,78 @@ def availability_errors(self, resource):
 
-    def __init__(self, package_name, required_version=None):
+    def __init__(self, app_name, required_version, version_opt, version_regex):
         """ Initializer.
 
-            :param str package_name: the name of the package
-            :param required_version: the required version for this package
+            :param str app_name: the name of the application
+            :param required_version: the required version for this application
+            :param version_opt: command line option to print version of this application
+            :param version_regex: regular expression to extract version from
+                                  output of @version_opt
             :type required_version: :class:`distutils.LooseVersion` or NoneType
         """
-        self.package_name = package_name
+        self.app_name = app_name
         self.required_version = required_version
+        self.version_opt = version_opt
+        self.version_regex = version_regex
 
     def __str__(self):
-        return "%s-%s" % (self.package_name, self.required_version)
+        return "%s-%s" % (self.app_name, self.required_version)
 
 
-class PackageMethod(Method):
+class VersionMethod(Method):
 
-    """ Methods for checking the package version of the external resource. """
+    """ Methods for checking the version of the external resource. """
 
-    def __init__(self, package=None):
+    def __init__(self, version_info=None):
         """ Initializer.
 
-            :param :class:`PackageInfo` package:
+            :param :class:`AppVersionInfo` version_info:
         """
-        self.package = package
+        self.version_info = version_info
         self._availability_errors = None
 
     @property
-    def package_version(self):
-        """ Returns the version of the installed package.
+    def app_version(self):
+        """ Returns the version of the installed application.
 
-            :returns: the package version
+            :returns: the application version
             :rtype: LooseVersion
-            :raises AvailabilityError: on failure to obtain package version
+            :raises AvailabilityError: on failure to obtain application version
         """
-        sack = hawkey.Sack()
 
         try:
-            sack.load_system_repo()
-        except IOError as e:
-            # hawkey has been observed allowing an IOError to propagate to
-            # caller with message "Failed calculating RPMDB checksum."
-            # See: https://bugzilla.redhat.com/show_bug.cgi?id=1223914
-            raise AvailabilityError("Could not determine package version for %s: %s" % (self.package.package_name, e))
+            out = capture_output([self.version_info.app_name,
+                                  self.version_info.version_opt],
+                                 stderr_to_stdout=True)
+        except OSError as e:
+            raise AvailabilityError("Could not determine version for %s: %s" % (self.version_info.app_name, e))
```

Comment:
Will this work if the application exits with a code != 0? And is there any reason to not use the ``utils_check_util_version()`` function from *libblockdev*?